app/src/main/kotlin/com/intive/picover/validators/ValidationStatus.kt

```diff
@@ -0,0 +1,33 @@
+package com.intive.picover.validators
+
+import com.intive.picover.R
+
+sealed class ValidationStatus {
+	object EmptyText : ValidationStatus()
+	object BlankText : ValidationStatus()
+	object TooLongText : ValidationStatus()
+	object ValidText : ValidationStatus()
+
+	fun errorMessageResId(): Int {
+		when (this) {
+			is EmptyText -> {
+				return R.string.TextShouldNotNeEmpty
+			}
+
+			is BlankText -> {
+				return R.string.TextShouldNotBeBlank
+			}
+
+			is TooLongText -> {
+				return R.string.TextIsTooLong
+			}
+
+			else -> {
+				return R.string.TextIsValid
+			}
+		}
+	}
```

Comment:
minor: instead of `when` you could use polymorphism ;)

Reply:
Done 😄